diff --git a/mysql-test/CMakeLists.txt b/mysql-test/CMakeLists.txt
index 7c1e82b..a92ba91 100644
--- a/mysql-test/CMakeLists.txt
+++ b/mysql-test/CMakeLists.txt
@@ -56,6 +56,10 @@ INSTALL(
 ENDIF()
 
 
+# Expand some paths in the perl scripts correctly
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/lib/My/ConfigFactory.pm ${CMAKE_CURRENT_SOURCE_DIR}/lib/My/ConfigFactory.pm @ONLY)
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/mysql-test-run.pl ${CMAKE_CURRENT_SOURCE_DIR}/mysql-test-run.pl @ONLY)
+
 IF(NOT ${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
   # Enable running mtr from build directory
   CONFIGURE_FILE(
diff --git a/mysql-test/mysql-test-run.pl b/mysql-test/mysql-test-run.pl
index e9130dd..388c039 100755
--- a/mysql-test/mysql-test-run.pl
+++ b/mysql-test/mysql-test-run.pl
@@ -1535,11 +1535,13 @@ sub command_line_setup {
   }
 
   # Look for language files and charsetsdir, use same share
-  $path_language= mtr_path_exists("$bindir/share/mysql",
+  $path_language= mtr_path_exists("$bindir/@INSTALL_MYSQLSHAREDIR@",
+                                  "$bindir/share/mysql",
                                   "$bindir/share");
   my $path_share= $path_language;
 
-  @share_locations= ("share/mysql",
+  @share_locations= ("@INSTALL_MYSQLSHAREDIR@",
+                     "share/mysql",
                      "share/mysql-" . $mysql_base_version,
                      "share");
 
@@ -4140,6 +4142,7 @@ sub mysql_install_db {
 
   my $path_sql= my_find_file($install_basedir,
 			     ["mysql", "share/mysql",
+			      "@INSTALL_MYSQLSHAREDIR@",
 			      "share/mysql-" . $mysql_base_version,
 			      "share", "scripts"],
 			      "mysql_system_tables.sql",
