diff -up mysql-5.7.16/cmake/install_layout.cmake.mysqlfilesdir mysql-5.7.16/cmake/install_layout.cmake
--- mysql-5.7.16/cmake/install_layout.cmake.mysqlfilesdir	2016-10-21 18:42:47.501440490 +0200
+++ mysql-5.7.16/cmake/install_layout.cmake	2016-10-21 18:45:34.476521200 +0200
@@ -142,7 +142,7 @@ ELSEIF(INSTALL_LAYOUT MATCHES "RPM" OR
        INSTALL_LAYOUT MATCHES "SLES" OR
        INSTALL_LAYOUT MATCHES "SVR4" OR
        INSTALL_LAYOUT MATCHES "DEB")
-  SET(secure_file_priv_path "/var/lib/mysql-files")
+  SET(secure_file_priv_path "${LOCALSTATEDIR}/mysql-files")
 ELSE()
   SET(secure_file_priv_path "${default_prefix}/mysql-files")
 ENDIF()
