diff -up mysql-5.5.36/mysql.init.scl-env-check mysql-5.5.36/mysql.init
--- mysql-5.5.36/mysql.init.scl-env-check	2014-02-12 20:30:35.948926604 +0100
+++ mysql-5.5.36/mysql.init	2014-02-13 09:28:21.670842074 +0100
@@ -1,7 +1,7 @@
 #!/bin/sh
 #
-# mysqld	This shell script takes care of starting and stopping
-#		the MySQL subsystem (mysqld).
+# __SCL_NAME__-mysqld	This shell script takes care of starting and stopping
+#                       the MySQL subsystem (mysqld).
 #
 # chkconfig: - 64 36
 # description:	MySQL database server.
@@ -9,7 +9,7 @@
 # config: /etc/my.cnf
 # pidfile: /var/run/mysqld/mysqld.pid
 ### BEGIN INIT INFO
-# Provides: mysqld
+# Provides: __SCL_NAME__-mysqld
 # Required-Start: $local_fs $remote_fs $network $named $syslog $time
 # Required-Stop: $local_fs $remote_fs $network $named $syslog $time
 # Short-Description: start and stop MySQL server
@@ -24,7 +24,7 @@
 
 
 exec="/usr/bin/mysqld_safe"
-prog="mysqld"
+prog="__SCL_NAME__-mysqld"
 
 # Set timeouts here so they can be overridden from /etc/sysconfig/mysqld
 STARTTIMEOUT=120
@@ -233,6 +233,18 @@ condrestart(){
     [ -e $lockfile ] && restart || :
 }
 
+# We have to re-enable SCL environment, because /sbin/service
+# clears almost all environment variables.
+# Since X_SCLS is cleared as well, we lose information about other
+# collections enabled.
+source __SCL_SCRIPTS__/service-environment
+source scl_source enable $MYSQL55_SCLS_ENABLED
+
+# we want start daemon only inside "scl enable" invocation
+if ! scl_enabled __SCL_NAME__ ; then
+    echo "Collection __SCL_NAME__ has to be listed in __SCL_SCRIPTS__/service-environment"
+    exit 1
+fi
 
 # See how we were called.
 case "$1" in
diff -up mysql-5.5.36/scripts/mysqld_safe.sh.scl-env-check mysql-5.5.36/scripts/mysqld_safe.sh
--- mysql-5.5.36/scripts/mysqld_safe.sh.scl-env-check	2014-02-13 09:28:21.628842087 +0100
+++ mysql-5.5.36/scripts/mysqld_safe.sh	2014-02-13 09:28:21.671842073 +0100
@@ -11,6 +11,12 @@
 # mysql.server works by first doing a cd to the base directory and from there
 # executing mysqld_safe
 
+# we want start daemon only inside "scl enable" invocation
+if ! scl_enabled __SCL_NAME__ ; then
+    echo "Use \"scl enable __SCL_NAME__ 'service ...'\" invocation"
+    exit 1
+fi
+
 # Initialize script globals
 KILL_MYSQLD=1;
 MYSQLD=
diff -up mysql-5.5.36/scripts/mysql_install_db.sh.scl-env-check mysql-5.5.36/scripts/mysql_install_db.sh
--- mysql-5.5.36/scripts/mysql_install_db.sh.scl-env-check	2014-01-14 09:16:26.000000000 +0100
+++ mysql-5.5.36/scripts/mysql_install_db.sh	2014-02-13 09:45:11.864672754 +0100
@@ -18,6 +18,12 @@
 #
 # All unrecognized arguments to this script are passed to mysqld.
 
+# we want start daemon only inside "scl enable" invocation
+if ! scl_enabled __SCL_NAME__ ; then
+    echo "Use \"scl enable __SCL_NAME__ 'service ...'\" invocation"
+    exit 1
+fi
+
 basedir=""
 builddir=""
 ldata="@localstatedir@"
@@ -410,15 +416,15 @@ else
   echo
   echo "You can try to start the mysqld daemon with:"
   echo
-  echo "    shell> $mysqld --skip-grant &"
+  echo "    shell> su -s /bin/bash - mysql -c 'scl enable __SCL_NAME__ -- $mysqld --skip-grant --general-log &'"
   echo
   echo "and use the command line tool $bindir/mysql"
   echo "to connect to the mysql database and look at the grant tables:"
   echo
-  echo "    shell> $bindir/mysql -u root mysql"
+  echo "    shell> scl enable __SCL_NAME__ -- mysql -u root mysql"
   echo "    mysql> show tables"
   echo
-  echo "Try 'mysqld --help' if you have problems with paths.  Using --log"
+  echo "Try 'scl enable __SCL_NAME__ -- mysqld --help' if you have problems with paths.  Using --log"
   echo "gives you a log in $ldata that may be helpful."
   echo
   echo "Please consult the MySQL manual section"
@@ -447,19 +453,15 @@ fi
 # the screen.
 if test "$cross_bootstrap" -eq 0 && test -z "$srcdir"
 then
-  s_echo
-  s_echo "To start mysqld at boot time you have to copy"
-  s_echo "support-files/mysql.server to the right place for your system"
-
   echo
   echo "PLEASE REMEMBER TO SET A PASSWORD FOR THE MySQL root USER !"
   echo "To do so, start the server, then issue the following commands:"
   echo
-  echo "$bindir/mysqladmin -u root password 'new-password'"
-  echo "$bindir/mysqladmin -u root -h $hostname password 'new-password'"
+  echo "scl enable __SCL_NAME__ -- $bindir/mysqladmin -u root password 'new-password'"
+  echo "scl enable __SCL_NAME__ -- $bindir/mysqladmin -u root -h $hostname password 'new-password'"
   echo
   echo "Alternatively you can run:"
-  echo "$bindir/mysql_secure_installation"
+  echo "scl enable __SCL_NAME__ -- $bindir/mysql_secure_installation"
   echo
   echo "which will also give you the option of removing the test"
   echo "databases and anonymous user created by default.  This is"
@@ -471,10 +473,11 @@ then
   then
     echo
     echo "You can start the MySQL daemon with:"
-    echo "cd $basedir ; $bindir/mysqld_safe &"
+    echo "service __SCL_NAME__-mysqld start"
     echo
     echo "You can test the MySQL daemon with mysql-test-run.pl"
-    echo "cd $basedir/mysql-test ; perl mysql-test-run.pl"
+    echo "after installing __SCL_NAME__-mysql-test package."
+    echo "See @prefix@/share/mysql-test/README for instructions."
   fi
 
   echo
